src/main/java/com/rarchives/ripme/ripper/rippers/XhamsterRipper.java

```diff
@@ -156,7 +156,7 @@ public Document getNextPage(Document doc) throws IOException {
         if (!isVideoUrl(url)) {
             if (!doc.select("div.picture_view > div.pictures_block > div.items > div.item-container > a.item").isEmpty()) {
                 // Old HTML structure is still present at some places
-                for (Element page : doc.select("div.picture_view > div.pictures_block > div.items > div.item-container > a.item")) {
+                for (Element page : doc.select(".clearfix > div > a.slided")) {
```

Comment:
Is it possible to get a more unique XPath?

Reply:
tbh i am not really using ripme apart from rare occasions for instagram, so if you d be able to propse a better one i'd be glad. otherwise if this one improves the situation we should merge it, and improve it further if it if somebody reports an issue with a clear example.